lib/stripe/stripe_object.rb

```diff
@@ -142,73 +146,50 @@ def respond_to?(symbol)
       end
     end
 
-    def serialize_nested_object(key)
-      new_value = @values[key]
-      if new_value.is_a?(APIResource)
-        return {}
-      end
-
-      if @unsaved_values.include?(key)
-        # the object has been reassigned
-        # e.g. as object.key = {foo => bar}
-        update = new_value
-        update = update.to_hash if update.is_a?(StripeObject)
-        new_keys = update.keys.map(&:to_sym)
-
-        # remove keys at the server, but not known locally
-        if @original_values[key]
-          keys_to_unset = @original_values[key].keys - new_keys
-          keys_to_unset.each {|key| update[key] = ''}
-        end
-
-        update
-      else
-        # can be serialized normally
-        self.class.serialize_params(new_value)
+    # Sets all keys within the StripeObject as unsaved so that they will be
+    # included with an update when #serialize_params is called. This method is
+    # also recursive, so any StripeObjects contained as values or which are
+    # values in a tenant array are also marked as dirty.
+    def dirty!
+      @unsaved_values = Set.new(@values.keys)
+      @values.each do |k, v|
+        dirty_value!(v)
       end
     end
 
-    def self.serialize_params(obj, original_value=nil)
-      case obj
-      when nil
-        ''
-      when Array
-        update = obj.map { |v| serialize_params(v) }
-        if original_value != update
-          update
-        else
-          nil
+    def serialize_params(options = {})
+      update_hash = {}
+
+      @values.each do |k, v|
+        # There are two reasons we may want to add in a parameter for update:
```

Comment:
Oops. I just removed the number so that this mistake won't happen again. Thanks!